tests/TransactionBuilderTest.php

```diff
@@ -0,0 +1,220 @@
+<?php
+declare(strict_types=1);
+
+use Illuminate\Support\Facades\DB;
+
+class TransactionBuilderTest extends TestCase
+{
+    protected $insertData = ['name' => 'klinson', 'age' => 20, 'title' => 'admin'];
+    protected $originData = ['name' => 'users', 'age' => 20, 'title' => 'user'];
+    protected $connection = 'mongodb_repl';
+    protected $originConnection = 'mongodb';
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        /** change connection to seplset? because the transaction needs  */
+        $this->originConnection = DB::getDefaultConnection();
+        DB::setDefaultConnection($this->connection);
+
+        DB::collection('users')->truncate();
+        DB::collection('users')->insert($this->originData);
+    }
+
+    protected function getEnvironmentSetUp($app)
+    {
+        if (version_compare(env('MONGO_VERSION'), '4', '<')) {
+            $this->markTestSkipped('MongoDB with version below 4 is not supported for transactions');
+        }
+
+        $config = require 'config/database.php';
+
+        $app['config']->set('database.connections.'.$this->connection, $config['connections'][$this->connection]);
+        $app['config']->set('database.default', $this->connection);
+    }
+
+    public function tearDown(): void
+    {
+        DB::collection('users')->truncate();
+        DB::setDefaultConnection($this->originConnection);
+        parent::tearDown();
+    }
+
+    public function testInsert()
+    {
+        /** rollback test */
+        DB::beginTransaction();
+        DB::collection('users')->insert($this->insertData);
```

Comment:
I think these tests would be much more readable if they didn't reference a class-level property for the data fixture(s).

I'd even say that the original data (i.e. `$originData`) is small enough that it could be inlined within each method that needs it. There's no real need to use the exact same fixture data in each test, and using literal values would be clearer than code like `$this->insertData['name'])->get()` and `$this->originData['age'] + 1`.

Reply:
Done